dataproc/create_hail_cluster.ipynb

```diff
@@ -0,0 +1,865 @@
+   "id": "6f5b9f2d-41b7-497c-b92d-f5449c8e3f12",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from datetime import datetime\n",
+    "import os"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e7ba8803-aa38-407a-a4a2-6aa5eda6bd7b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if not (os.getenv('GOOGLE_CLOUD_PROJECT')\n",
+    "        and os.getenv('GOOGLE_SERVICE_ACCOUNT_EMAIL')\n",
+    "        and os.getenv('TERRA_USER_EMAIL')):\n",
+    "    raise Exception('Expected environment variables are not available. Please let terra-support@verily.com know.')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14b47701-4178-45e6-8ed9-3f65a4bbf57e",
+   "metadata": {},
+   "source": [
+    "Obtain the GCP project ID so that it can be used in the name of any buckets we create and also to tell Dataproc where to create the cluster."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bf97da2f-b23c-46dd-af66-f2738dda8472",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "PROJECT = os.getenv('GOOGLE_CLOUD_PROJECT')\n",
+    "\n",
+    "PROJECT"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d145af3b-80b8-4457-85a0-e3ef55213475",
+   "metadata": {},
+   "source": [
+    "Obtain the Enterprise Terra user's service account. They will act as this same service account when running notebooks or batch scripts on the Dataproc cluster."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "78203c21-40bf-4b8a-b2f5-1d352fa1319e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "SERVICE_ACCOUNT = os.getenv('GOOGLE_SERVICE_ACCOUNT_EMAIL')\n",
+    "\n",
+    "SERVICE_ACCOUNT"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "49537cb3-20b4-4135-b316-efb95a995172",
+   "metadata": {},
+   "source": [
+    "Obtain the user name so that it can become part of the Hail cluster name and staging bucket. This is useful when people collaborate in Enterprise Terra workspaces and want to differentiate their clusters from each other."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "47735c99-12d4-47c2-b2ae-2e21da4f73cf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "USER = os.getenv('TERRA_USER_EMAIL').split('@')[0].replace('.', '-')\n",
+    "\n",
+    "USER"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1133b444-a427-4a72-a92b-0acb2c146844",
+   "metadata": {},
+   "source": [
+    "## Check the Enterprise Terra reference to the Dataproc staging bucket\n",
+    "\n",
+    "Create the user's Dataproc staging bucket, if it does not yet exist."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1e9af1d8-64f4-4d1a-ba32-e5eb4eff6565",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!terra resource resolve --name dataproc_staging_{USER} || terra resource create gcs-bucket --name=dataproc_staging_{USER} \\\n",
```

Comment:
nit: line is a bit too long, consider adding slash-newline so the bash code displays well on GitHub